src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -321,27 +325,24 @@ private void Initialize(XmlReader reader, string fileName, bool ignoreErrors)
             catch (Exception exception)
             {
                 InitializeSucceeded = false;
-                if (exception.MustBeRethrown())
+
+                if (exception.IsServereException())
                 {
                     throw;
                 }
 
-                NLogConfigurationException ConfigException = new NLogConfigurationException("Exception occurred when loading configuration from " + fileName, exception);
+                var message = string.Format("Exception occurred when loading configuration from '{0}': {1}", fileName, exception);
 
-                if (!ignoreErrors)
+                if (ignoreErrors)
                 {
-                    if (LogManager.ThrowExceptions)
-                    {
-                        throw ConfigException;
-                    }
-                    else
-                    {
-                        InternalLogger.Error("Error in Parsing Configuration File. Exception : {0}", ConfigException);
-                    }
+                    InternalLogger.Error(message);
                 }
                 else
                 {
-                    InternalLogger.Error("Error in Parsing Configuration File. Exception : {0}", ConfigException);
+                    if (exception.MustBeRethrown(message))
+                    {
+                        throw new NLogConfigurationException(message);
+                    }
                 }
             }
         }
```

Comment:
Ok, then.

The only remaining problem that I see here, then, is that `message` above includes the `exception`. So when we call `MustBeRethrown(message)`, `MustBeRethrown()` will add the exception again, so the exception text will appear twice in the internal log.